### ISSUE TITLE
fix wrong data access

### ISSUE THREAD

**Issue by BenjaminRodenberg** [BenjaminRueth/waveform-bindings#3]
https://github.com/BenjaminRueth/waveform-bindings/blob/988cb2f316a9526b29ca470ea47a6a4e880a48a8/waveformbindings/waveformbindings.py#L108-L112

Should be something like this:

```
if self._write_info["data_dimension"] == 1:
        write_data = write_waveform._samples_in_time[:, substep]
        super().write_block_scalar_data(write_data_id, self._write_info["vertex_ids"], write_data)
elif self._write_info["data_dimension"] == self.get_dimensions():
        write_data = write_waveform._samples_in_time[:, :, substep]
        super().write_block_vector_data(write_data_id, self._write_info["vertex_ids"], write_data)
```

**Comment by BenjaminRodenberg** [comment]
Maybe even a cleaner solution: use the sample function and avoid direct access to `_samples_in_time`.

**Comment by BenjaminRueth** [comment]
partially fixed in #4, let's still keep this issue until https://github.com/BenjaminRueth/waveform-bindings/issues/3#issuecomment-542147002 is implemented

**Comment by BenjaminRodenberg** [comment]
Partially fixed in #4 . let's keep the issue open until https://github.com/BenjaminRueth/waveform-bindings/issues/3#issuecomment-542147002 is implemented
